### PR TITLE
BUG: The setting xrot=0 in DataFrame.hist() doesn't work with by and subplots #30288

### LINKED ISSUE #30288: The setting xrot=0 in DataFrame.hist() doesn't work with by and subplots
#### Code Sample, a copy-pastable example if possible

```python
df = pd.DataFrame({
    'length': [1.5, 0.5, 1.2, 0.9, 3],
    'animal': ['pig', 'rabbit', 'pig', 'pig', 'rabbit']
})
df.hist(column='length', by='animal', bins=5, xrot=0);
```
#### Problem description

The setting `xrot=0` doesn't work when I use the option `by` in DataFrame.hist() (see the screenshot below). The temporary workaround is to set it to a very small value like `xrot=0.001`.

image

#### Expected Output

Should look like this instead
image

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 3.10.0-1062.4.1.el7.x86_64
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.2
numpy            : 1.17.2
pytz             : 2019.3
dateutil         : 2.8.0
pip              : 19.3.1
setuptools       : 41.4.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.8.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : 3.0.2
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.3.1
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by souvik3333**
- [x] closes #30288
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/plotting/test_hist_method.py [hunk 1]
```diff
@@ -253,6 +253,23 @@ def test_tight_layout(self):
 
         tm.close()
 
+    def test_hist_subplot_xrot(self):
+        df = DataFrame(
```
Comment (jreback) [inline pandas/tests/plotting/test_hist_method.py]: can you add the issue number here as a comment

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew note (plotting bugs)

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. pls add a whatsnew note in plotting bugfixes for 1.0.0, ping on green.

File: pandas/tests/plotting/test_hist_method.py (hunk 1 above)
Comment (souvik3333) [reply pandas/tests/plotting/test_hist_method.py]: Thanks @jreback. I have added the issue number.

**Comment by souvik3333** [comment]
@jreback, I have updated the whatsnew entry.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
great thanks @souvik3333

**Pull request merged**